field.go

```diff
@@ -467,6 +481,49 @@ func (f *Field) applyOptions(opt FieldOptions) error {
 	return nil
 }
 
+// loadAvailableShards reads remoteAvailableShards data for the field, if any.
+func (f *Field) loadAvailableShards() error {
+	bm := roaring.NewBitmap()
+
+	// Read data from meta file.
+	buf, err := ioutil.ReadFile(filepath.Join(f.path, ".available.shards"))
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return errors.Wrap(err, "reading available shards")
+	} else {
+		if err := bm.UnmarshalBinary(buf); err != nil {
+			return errors.Wrap(err, "unmarshaling")
+		}
+	}
+
+	// Merge bitmap from file into field.
+	f.mergeRemoteAvailableShards(bm)
+
+	return nil
+}
+
+// saveAvailableShards writes remoteAvailableShards data for the field.
+func (f *Field) saveAvailableShards() error {
+	f.mu.RLock()
+	defer f.mu.RUnlock()
+
+	// Write available shards to buffer.
+	var buf bytes.Buffer
+	if n, err := f.remoteAvailableShards.WriteTo(&buf); err != nil {
```

Comment:
why write to a buffer rather than directly to the file?

Reply:
good catch. i removed it in ee503b6ec0e31c1f5d976a2e8b01c204e5c41c13